leaf-server/minecraft-patches/features/0285-optimize-Entity-getOnPos.patch

```diff
@@ -0,0 +1,52 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: hayanesuru <hayanesuru@outlook.jp>
+Date: Mon, 4 Aug 2025 18:55:55 +0900
+Subject: [PATCH] optimize Entity#getOnPos
+
+
+diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
+index 15c5a90e77b955eac7f5f5fa40d5a0dba4c4619c..0f80680f6a673acf3a07abb0f920b1e9d81fd4af 100644
+--- a/net/minecraft/world/entity/Entity.java
++++ b/net/minecraft/world/entity/Entity.java
+@@ -1482,18 +1482,26 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
+     }
+ 
+     protected BlockPos getOnPos(float yOffset) {
+-        if (this.mainSupportingBlockPos.isPresent() && this.level().getChunkIfLoadedImmediately(this.mainSupportingBlockPos.get()) != null) { // Paper - ensure no loads
++        // Leaf start - optimize get chunk
++        if (this.mainSupportingBlockPos.isPresent()) {
+             BlockPos blockPos = this.mainSupportingBlockPos.get();
+             if (!(yOffset > 1.0E-5F)) {
+                 return blockPos;
+             } else {
+-                BlockState blockState = this.level().getBlockState(blockPos);
+-                return (!(yOffset <= 0.5) || !blockState.is(BlockTags.FENCES))
+-                        && !blockState.is(BlockTags.WALLS)
+-                        && !(blockState.getBlock() instanceof FenceGateBlock)
++                BlockState blockState = this.level().getBlockStateIfLoadedUnchecked(blockPos.getX(), blockPos.getY(), blockPos.getZ());
++                if (blockState == null) {
++                    int floor = Mth.floor(this.position.x);
++                    int floor1 = Mth.floor(this.position.y - yOffset);
++                    int floor2 = Mth.floor(this.position.z);
++                    return new BlockPos(floor, floor1, floor2);
++                }
++                return (!(yOffset <= 0.5) || ((blockState.tagFlag & 2) != 2))
```

Comment:
Well teeeeeeeeeeechnically, if you know the result is "something or 0" it's usually better to check != 0 or == 0, because then the machine code will check the zero flag (`jz` instruction) instead of doing an == comparison, which is potentially faster.